packages/ripple-binary-codec/src/binary.ts

```diff
@@ -105,16 +106,19 @@ interface ClaimObject extends JsonObject {
  * @returns the serialized object with appropriate prefix
  */
 function signingClaimData(claim: ClaimObject): Buffer {
-  const num = bigInt(String(claim.amount))
   const prefix = HashPrefix.paymentChannelClaim
   const channel = coreTypes.Hash256.from(claim.channel).toBytes()
-  const amount = coreTypes.UInt64.from(num).toBytes()
-
   const bytesList = new BytesList()
-
   bytesList.put(prefix)
   bytesList.put(channel)
-  bytesList.put(amount)
+  if (typeof claim.amount === 'object') {
```

Comment:
This is a nit but I think the code would be easier to read if the if-else blocks were flipped - `if (typeof claim.amount === 'string')` etc